### PR TITLE
tests: port document-portal-activation to session-tool

### PR DESCRIPTION
**Pull request by zyga**
There are a number of usual changes, move to session-tool, not leaking
dbus-daemon, running a session service as the test user. That last
change mainly helps to make test cleanup reliable. I also did three
extra changes along the way:

 - The check_stderr helper no longer checks cgroup type each time.
 - The helper variables were removed, this makes some commands longer
   but more understandable and less quirky to quoting and shell
   expansion issues.
 - The variant where session bus is unavailable is now tested last,
   with session bus socket stopped and with environment override
   that stops snap run from trying to talk to a non-existent socket.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: tests/main/document-portal-activation/task.yaml [hunk 1]
```diff
@@ -17,75 +17,75 @@ description: |
...
-    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
-    check_stderr stderr.log
-
-    echo "Starting session bus"
-    eval "$(dbus-launch --sh-syntax)"
-    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+    if is_cgroupv2; then
+        check_stderr() {
+            test -n "$1"
+            MATCH "WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement" < "$1"
```
Comment (bboozzoo) [inline tests/main/document-portal-activation/task.yaml:53]: I think this is incorrect compared to the previous code. The removed code checked that the warning is the only content of the stderr file. In the current code the check is still true when the warning is followed by some random content.

Comment (zyga) [reply tests/main/document-portal-activation/task.yaml:53]: Ah, good point. I'll adjust this.

Comment (zyga) [reply tests/main/document-portal-activation/task.yaml]: Done

File: tests/main/document-portal-activation/task.yaml [hunk 2]
```diff
@@ -17,75 +17,75 @@ description: |
...
+    fi
 
     echo "No output on stderr when running with a session bus, when xdg-document-portal is not present."
-    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
+    session-tool -u test test-snapd-desktop.check-dirs /home/test/snap/test-snapd-desktop/current 2>stderr.log
     check_stderr stderr.log
 
     echo "The absence of the document portal service was recorded"
-    [ -f "$XDG_RUNTIME_DIR/.portals-unavailable" ]
+    test -f /run/user/12345/.portals-unavailable
```
Comment (stolowski) [inline tests/main/document-portal-activation/task.yaml:71]: Is there a preference for `test` over square bracket op? Just curious.

File: tests/main/document-portal-activation/task.yaml [hunk 3]
```diff
@@ -17,75 +17,75 @@ description: |
 # Disabled on Ubuntu Core because it doesn't provide the "desktop"
 # slot, and Amazon Linux because it doesn't have the required Python 3
 # packages to run the test.
-systems: [ -ubuntu-core-*, -amazon-linux-2-*, -centos-* ]
-
-environment:
-    XDG_RUNTIME_DIR: /run/user/$(id -u)
```
Comment (stolowski) [inline tests/main/document-portal-activation/task.yaml:23]: Would it make sense to have a variable for /run/user/12345/ instead?

File: tests/main/document-portal-activation/task.yaml (hunk 2 above)
Comment (zyga) [reply tests/main/document-portal-activation/task.yaml:71]: This is my personal preference:

I will use `[` in an if statement:

```
if [ "$stuff" = foo ]; then
   ...
fi
```
This looks like regular conditional statement in most languages.

I will use `test` when it acts as an assertion.
```
test "$stuff = foo
````
This looks like C `assert()`

File: tests/main/document-portal-activation/task.yaml (hunk 3 above)
Comment (zyga) [reply tests/main/document-portal-activation/task.yaml:23]: I personally don't think so, it tends to be quirky or annoying with quoting, necessitating the the use of shellcheck disable comments. It also is heavier cognitively, requiring the reader to jump back and forth.

In addition the variable should never have been called `XDG_RUNTIME_DIR`, as that interferes with legitimate use of the variable already present in the `bash` process executing the test and can lead to confusing and hard-to-debug failures.

**Pull request merged**